onnxruntime/core/platform/posix/env.cc

```diff
@@ -220,16 +226,22 @@ class PosixThread : public EnvThread {
     std::unique_ptr<Param> p(static_cast<Param*>(param));
     ORT_TRY {
 #if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
-      if (p->affinity_mask.has_value()) {
+      if (p->affinity.has_value() && !p->affinity->logical_proc_ids.empty()) {
         cpu_set_t cpuset;
         CPU_ZERO(&cpuset);
-        CPU_SET(*p->affinity_mask, &cpuset);
+        for(auto id : p->affinity->logical_proc_ids) {
+          CPU_SET(id, &cpuset);
+        }
         // pthread_setaffinity_np() does not set errno, it returns it.
         auto ret = pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
         if (ret != 0) {
           auto [err_no, err_msg] = GetSystemError(ret);
-          LOGS_DEFAULT(ERROR) << "pthread_setaffinity_np failed for thread: " << pthread_self()
-                              << ", mask: " << *p->affinity_mask
+          const auto masks = Env::Default().GetThreadAffinityMasks();
+          std::ostringstream os;
+          os << *p->affinity;
```

Comment:
do we need an intermediate std::ostringstream?

Reply:
Will try without it